pymc_marketing/mlflow.py

```diff
@@ -577,8 +941,35 @@ def new_fit(*args, **kwargs):
                 "saturation_name",
                 json.loads(idata.attrs["saturation"])["lookup_name"],
             )
+
+            # Align with the default values in pymc.sample
+            tune = kwargs.get("tune", 1000)
+
+            if log_sampler_info:
+                log_sample_diagnostics(idata, tune=tune)
+                log_arviz_summary(
+                    idata,
+                    "summary.html",
+                    var_names=summary_var_names,
+                    **arviz_summary_kwargs,
+                )
+
             log_inference_data(idata, save_file="idata.nc")
 
+            if log_loocv:
+                log_loocv_metrics(idata=idata)
+
+            posterior_preds = self.sample_posterior_predictive(self.X)
+            log_summary_metrics(
+                y_true=self.y,
+                y_pred=posterior_preds[self.output_var],
+            )
```

Comment:
Don't know how I feel about the `fit` method also calling the `sample_posterior_predictive`. Seems a bit surprising to me. Does this restrict calling the `sample_posteior_predictive` a second time? Maybe patch the `sample_posterior_predictive` method with this logic. That puts a wrench in the `log_model` though. (Typing out load)

Thoughts here? CC @juanitorduz?

Reply:
I agree with your suggestion Will :)

Reply:
Ok yep, makes sense - I just separated it into its own patched function, hopefully that's better now 😁